playbooks/operationalizing/secure-forward-splunk-container.adoc

```diff
@@ -0,0 +1,269 @@
+** A preconfigured Splunk Docker image to use for testing can be found link:https://github.com/themoosman/docker-splunk/tree/master/enterprise[here]
+* An OpenShift Container Platform cluster with aggregated logging installed
+
+==== Note
+If using the preconfigured Docker image, this section can be skipped as these changes are already completed in the image.
+
+=== Configure Splunk
+
+To accept messages sent from the fluent-plugin-splunk-ex plugin, a new TCP based data input must be configured within Splunk.
+
+As a Splunk user with administrative privileges, navigate to the Splunk user interface. Select *Settings* -> *Data Inputs* and then within the _Local Inputs_ section, select *Add new* next to the _TCP_ type.
+
+Specify the following values to configure the TCP based Splunk input:
+
+* Port: 9997
+* Source name override: OCP
+* Source Type: _json
+
+NOTE: It is beyond the scope of this document for the creation of a custom link:https://docs.splunk.com/Splexicon%3ASourcetype[sourcetype] to perform advanced parsing of the data received.
+
+Complete the necessary fields to finalize the addition of the new TCP input
+
+NOTE: Open applicable firewall ports on the Splunk machine to allow for data to be received
+
+Since the fluent-plugin-splunk-ex plugin sends data to Splunk in batches, Splunk must be configured to separate messages based on the presence of a new line character.
+
+Configure the _/opt/splunk/apps/search/local/props.conf_ file with the following content which will configure messages received from the _OCP_ source to split messages correctly:
+
+[source]
+----
+[source::OCP]
+SHOULD_LINEMERGE = false
+----
+
+Restart Splunk to apply the changes:
+
+[source]
+----
+systemctl restart splunk
+----
+
+
+=== Containerized Fluentd
+
+A standalone instance of Fluentd serves as a broker between the OpenShift aggregated logging solution and Splunk. It receives messages sent by the secure forward plugin and forwards them to Splunk using the link:https://github.com/gtrevg/fluent-plugin-splunk-ex[fluent-plugin-splunk-ex] plugin.
+In this implementation, the "external" fluentd instance will be run in OpenShift.
+
+==== Source
+The source code for the implementation can be found link:https://github.com/themoosman/openshift-fluentd-forwarder[here].
+
+==== Create Fluentd Build Configuration
+
+Create the fluentd-forwarder build configuration in the logging project.
+
+[source]
+----
+oc project logging && \
+oc process -f fluentd-forwarder-build-config-template.yaml \
+-p GIT_URI=https://github.com/themoosman/openshift-fluentd-forwarder.git | oc create -f
+----
+
+Build the Docker image in OpenShift and monitor for any errors.
+
+[source]
+----
+oc start-build fluentd-forwarder-build --follow=true
+----
+
+
+==== Create Fluentd Deployment Configuration
+
+Create the fluentd-forwarder deployment configuration in the logging project.
+
+[source]
+----
+oc project logging && \
+oc process -f fluentd-forwarder-template.yaml \
```

Comment:
Template does not specify that the `ImageChangeTrigger` is `automatic: true` so you have to roll it out manually